util/buildinfo/buildinfo.go

```diff
@@ -136,6 +143,60 @@ func mergeSources(ctx context.Context, buildSources map[string]string, frontendS
 	return srcs, nil
 }
 
+// decodeDeps decodes dependencies (buildinfo) added via the input context.
+func decodeDeps(key string, attrs map[string]*string) (map[string]binfotypes.BuildInfo, error) {
+	var platform string
+	// extract platform from metadata key
+	skey := strings.SplitN(key, "/", 2)
+	if len(skey) == 2 {
+		platform = skey[1]
+	}
+
+	res := make(map[string]binfotypes.BuildInfo)
+	for k, v := range attrs {
+		// dependencies are only handled via the input context
+		if v == nil || !strings.HasPrefix(k, "input-metadata:") {
+			continue
+		}
+
+		// if platform is defined, only decode dependencies for that platform
+		if platform != "" && !strings.HasSuffix(k, "::"+platform) {
+			continue
```

Comment:
As discussed offline, we should do a follow-up that handles the contexts without suffixes as well for multi-platform cases.